engines/standard_tasks/app/models/standard_tasks/paper_reviewer_task.rb

```diff
@@ -6,6 +6,18 @@ def self.permitted_attributes
 
     register_task default_title: "Assign Reviewers", default_role: "editor"
 
+    include Invitable
+
+    def invitation_invited(invitation)
+      PaperReviewerMailer.delay.notify_invited({
+        invitation_id: invitation.id
+      })
+    end
+
+    def invitation_accepted(invitation)
+      TaskRoleUpdater.new(self, invitation.invitee_id, PaperRole::REVIEWER).update
+    end
+
     def array_attributes
       super + [:reviewer_ids]
```

Comment:
I know it's not directly related to what you're doing, but I wanted to float it by you.

Notice how this class is using the `reviewer_ids=` assignment to side effect a `PaperRole` into existence by updating the `PaperReviewerTask` in `TasksController#update`. I think we should get away from trying to do everything in the task. For example, have your client make an api call to create/delete the reviewer. For example, the `PaperEditorTask` was doing something very similar to this. I ended up moving the functionality to it's own controller (see `EditorsController`). Perhaps that could be expanded to handle paper role crud more completely. I decided to not jump in fully because the client had no concept of `PaperRole`, just `Editor`.

Anyway, something to think on...